src/Config.h

```diff
@@ -0,0 +1,1488 @@
+            _doc["hardware"]["sensors"]["pressure"]["enabled"] = value;
+        }
+
+        bool getWaterTankSensorEnabled() {
+            return _doc["hardware"]["sensors"]["watertank"]["enabled"] | false;
+        }
+
+        void setWaterTankSensorEnabled(const bool value) {
+            _doc["hardware"]["sensors"]["watertank"]["enabled"] = value;
+        }
+
+        int getWaterTankSensorMode() {
+            return _doc["hardware"]["sensors"]["watertank"]["mode"];
+        }
+
+        void setWaterTankSensorMode(const int value) {
+            _doc["hardware"]["sensors"]["watertank"]["mode"] = value;
+        }
+
+        // Scale
+        bool getScaleEnabled() {
+            return _doc["hardware"]["sensors"]["scale"]["enabled"] | false;
+        }
+
+        void setScaleEnabled(const bool value) {
+            _doc["hardware"]["sensors"]["scale"]["enabled"] = value;
+        }
+
+        int getScaleSamples() {
+            return _doc["hardware"]["sensors"]["scale"]["samples"];
+        }
+
+        void setScaleSamples(const int value) {
+            _doc["hardware"]["sensors"]["scale"]["samples"] = value;
+        }
+
+        int getScaleType() {
+            return _doc["hardware"]["sensors"]["scale"]["type"];
+        }
+
+        void setScaleType(const int value) {
+            _doc["hardware"]["sensors"]["scale"]["type"] = value;
+        }
+
+        float getScaleCalibration() {
+            return _doc["hardware"]["sensors"]["scale"]["calibration"] | static_cast<float>(SCALE_CALIBRATION_FACTOR);
+        }
+
+        void setScaleCalibration(const float value) {
+            _doc["hardware"]["sensors"]["scale"]["calibration"] = constrain(value, -100000.0f, 100000.0f);
+        }
+
+        void setScaleKnownWeight(const float value) {
+            _doc["hardware"]["sensors"]["scale"]["known_weight"] = constrain(value, 0.0f, 2000.0f);
+        }
+
+        float getScaleKnownWeight() {
+            return _doc["hardware"]["sensors"]["scale"]["known_weight"] | static_cast<float>(SCALE_KNOWN_WEIGHT);
+        }
+
+        float getScale2Calibration() {
+            return _doc["hardware"]["sensors"]["scale"]["calibration2"] | static_cast<float>(SCALE_CALIBRATION_FACTOR);
+        }
+
+        void setScale2Calibration(const float value) {
+            _doc["hardware"]["sensors"]["scale"]["calibration2"] = constrain(value, -100000.0f, 100000.0f);
+        }
+
+    private:
+        inline static auto CONFIG_FILE = "/config.json";
+
+        StaticJsonDocument<4096> _doc;
+
+        bool validateAndApplyConfig(const JsonDocument& doc) {
+            // PID parameters
+            if (doc["pid"].containsKey("enabled")) {
+                if (!doc["pid"]["enabled"].is<bool>()) {
+                    return false;
+                }
+
+                setPidEnabled(doc["pid"]["enabled"].as<bool>());
+            }
+
+            if (doc["pid"].containsKey("use_ponm")) {
```

Comment:
If I understand the code correctly, the JSON file is only ever accessed/created/altered through this class. IMHO it would then suffice to validate the parameters when setting and reading them instead of cycling through the entire file and re-implementing checks.